### PR TITLE
Add optional angle label to polygon

### PR DESCRIPTION
**Pull request by nedredmond**
## Summary
Adds optional angel label to Polygon. Isolates to its own component so that it can be reused for the Angle graph.

Points are always facing inside the graph.
Labels adjust their position when too close to the movable points.
Special arc for 90-degree angles.
Like original, omits "≈" when angle evaluates to integer.

https://github.com/Khan/perseus/assets/23404711/331dfb48-cbf0-4af6-8578-6fdb71e41859

### Issue
lems-1854

## Testing
Run `yarn dev`, flip mafs polygon flag, search "poly"

### CODE REVIEW

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:43]:
```suggestion
```

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:47]: I think there's a simpler way to calculate these `x1,y1,x2,y2` constants that doesn't need trigonometry:

```ts
const y1 = centerY + (startY - centerY) / a * radius;
const x2 = centerX + (endX - centerX) / b * radius;
const x1 = centerX + (startX - centerX) / a * radius;
const y2 = centerY + (endY - centerY) / b * radius;
```

With this approach, you don't need to calculate `startAngle` and `endAngle`.

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:24]: Color isn't used.

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:83]: What do these dimensions on the filter do? I tried deleting them and nothing seemed to change...

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:124]: When would `angle` ever be greater than pi? Could we just return `false` here, or omit this conditional?

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:127]: I think the tolerance on this should be smaller than `0.01`. 0.01 radians is ~0.5 degrees.
Screen Shot 2024-03-22 at 9 33 38 AM

File: packages/perseus/src/widgets/interactive-graphs/graphs/use-transform.ts
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graphs/use-transform.ts:9]: Nice!

File: packages/perseus/src/widgets/interactive-graphs/graphs/polygon.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graphs/polygon.tsx:59]: This return should be unreachable, right? I wonder if we should throw an exception here.

More likely than `pt1` or `pt2` being null is that they might be the same point, which I think will happen if there is only one point in the `points` array. In that case, I think we won't be able to compute an angle, so we probably should return null in that case.

**Review by benchristel — COMMENTED** [review]
This is pretty cool! I learned some math today.

It would be nice to have unit tests for all the math. I also left some suggestions inline.

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (nedredmond) [reply packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:82]: It adds 5% padding on the left and right of the filter.

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (nedredmond) [reply packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:123]: Radian math is wild, Ben. I learned a lot yesterday.

File: packages/perseus/src/widgets/interactive-graphs/graphs/polygon.tsx
Comment (nedredmond) [reply packages/perseus/src/widgets/interactive-graphs/graphs/polygon.tsx:59]: This could come in to play in "unlimited" mode.

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx
Comment (nedredmond) [reply packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx:24]: Removed!

**Review by benchristel — APPROVED** [review]
I think this is good to go if you've read through my suggestions and decided whether or not to implement them.

Although the bug about angle measurements we're discussing in Slack right now sounds like it might be a blocker. I'll leave fixing that up to you. Let me know if you want another review.

**Pull request merged**
